packages/core/src/routes/experience/classes/verifications/password-verification.ts

```diff
@@ -20,32 +20,43 @@ import { type IdentifierVerificationRecord } from './verification-record.js';
 export type PasswordVerificationRecordData = {
   id: string;
   type: VerificationType.Password;
-  identifier: InteractionIdentifier;
+  /** If user is not identified, identifier is required */
+  identifier?: InteractionIdentifier;
+  /** If user is already identified, userId is required */
+  userId?: string;
```

Comment:
userId should be a unique type of identifier?  In this case, a user is unverified but can verify the identity using `userId` and `password`.  `userId` is much like `username` here.